[ci.foundries.io] regression 4344e27 all: Update reserved function names

**Describe the bug**
Since the merge of 4344e27 all ARM based builds in sanitycheck have been failing the "tests/benchmarks/app_kernel/benchmark.application" test:

https://ci.foundries.io/projects/zephyr/builds/5287

The build break as can be sane for [blenano2](https://ci.foundries.io/projects/zephyr/builds/5287/sanity-compile-nrf52/artifacts/console.log) is:
~~~
[ 96%] Linking C executable zephyr_prebuilt.elf
cd /tmp/outdir/nrf52_blenano2/tests/benchmarks/app_kernel/benchmark.application/zephyr && /usr/local/bin/cmake -E cmake_link_script CMakeFiles/zephyr_prebuilt.dir/link.txt --verbose=1
ccache /opt/toolchains/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arm-zephyr-eabi/arm-zephyr-eabi-gcc    CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj  -o zephyr_prebuilt.elf -T linker.cmd -Wl,-Map=/tmp/outdir/nrf52_blenano2/tests/benchmarks/app_kernel/benchmark.application/zephyr/zephyr.map -u_OffsetAbsSyms -u_ConfigAbsSyms -Wl,--whole-archive ../app/libapp.a libzephyr.a arch/arch/arm/core/libarch__arm__core.a arch/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a arch/arch/arm/core/cortex_m/mpu/libarch__arm__core__cortex_m__mpu.a lib/libc/minimal/liblib__libc__minimal.a drivers/serial/libdrivers__serial.a -Wl,--no-whole-archive kernel/libkernel.a CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj -L"/opt/toolchains/zephyr-sdk-0.9.5/sysroots/armv5-zephyr-eabi/usr/lib/arm-zephyr-eabi/6.2.0/armv7e-m" -L/tmp/outdir/nrf52_blenano2/tests/benchmarks/app_kernel/benchmark.application/zephyr -lgcc -Wl,--print-memory-usage -mthumb -nostdlib -static -no-pie -Wl,--fatal-warnings -Wl,-X -Wl,-N -Wl,--gc-sections -Wl,--build-id=none -Wl,--orphan-handling=warn -mabi=aapcs 
Memory region         Used Size  Region Size  %age Used
           FLASH:       26496 B       512 KB      5.05%
            SRAM:       22732 B        64 KB     34.69%
        IDT_LIST:          40 B         2 KB      1.95lib/libc/minimal/liblib__libc__minimal.a(stdout_console.c.obj): In function `z_zephyr_fwrite':
/tmp/outdir/nrf52_blenano2/tests/benchmarks/app_kernel/benchmark.application/zephyr/include/generated/syscalls/libc-hooks.h:19: undefined reference to `z_impl_z_zephyr_fwrite'
~~~

--- a/include/misc/libc-hooks.h
+++ b/include/misc/libc-hooks.h
@@ -32,7 +32,7 @@ __syscall int z_zephyr_write_stdout(const void *buf, int nbytes);
 
 __syscall int _zephyr_fputc(int c, FILE *stream);
 
-__syscall size_t z_zephyr_fwrite(const void *_MLIBC_RESTRICT ptr, size_t size,
+__syscall size_t zephyr_fwrite(const void *_MLIBC_RESTRICT ptr, size_t size,
 				size_t nitems, FILE *_MLIBC_RESTRICT stream);
 #endif /* CONFIG_NEWLIB_LIBC */
 

--- a/include/misc/libc-hooks.h
+++ b/include/misc/libc-hooks.h
@@ -30,7 +30,7 @@ __syscall int z_zephyr_write_stdout(const void *buf, int nbytes);
 #else
 /* Minimal libc */
 
-__syscall int _zephyr_fputc(int c, FILE *stream);
+__syscall int zephyr_fputc(int c, FILE * stream);
 
 __syscall size_t zephyr_fwrite(const void *_MLIBC_RESTRICT ptr, size_t size,
 				size_t nitems, FILE *_MLIBC_RESTRICT stream);

--- a/lib/libc/minimal/source/stdout/stdout_console.c
+++ b/lib/libc/minimal/source/stdout/stdout_console.c
@@ -25,21 +25,21 @@ void __stdout_hook_install(int (*hook)(int))
 	_stdout_hook = hook;
 }
 
-int z_impl__zephyr_fputc(int c, FILE *stream)
+int z_impl_zephyr_fputc(int c, FILE *stream)
 {
 	return (stdout == stream) ? _stdout_hook(c) : EOF;
 }
 
 #ifdef CONFIG_USERSPACE
-Z_SYSCALL_HANDLER(_zephyr_fputc, c, stream)
+Z_SYSCALL_HANDLER(zephyr_fputc, c, stream)
 {
-	return z_impl__zephyr_fputc(c, (FILE *)stream);
+	return z_impl_zephyr_fputc(c, (FILE *)stream);
 }
 #endif
 
 int fputc(int c, FILE *stream)
 {
-	return _zephyr_fputc(c, stream);
+	return zephyr_fputc(c, stream);
 }
 
 int fputs(const char *_MLIBC_RESTRICT string, FILE *_MLIBC_RESTRICT stream)

--- a/lib/libc/minimal/source/stdout/stdout_console.c
+++ b/lib/libc/minimal/source/stdout/stdout_console.c
@@ -52,7 +52,7 @@ int fputs(const char *_MLIBC_RESTRICT string, FILE *_MLIBC_RESTRICT stream)
 	return len == ret ? 0 : EOF;
 }
 
-size_t z_impl__zephyr_fwrite(const void *_MLIBC_RESTRICT ptr, size_t size,
+size_t z_impl_zephyr_fwrite(const void *_MLIBC_RESTRICT ptr, size_t size,
 			    size_t nitems, FILE *_MLIBC_RESTRICT stream)
 {
 	size_t i;
@@ -82,11 +82,11 @@ done:
 }
 
 #ifdef CONFIG_USERSPACE
-Z_SYSCALL_HANDLER(z_zephyr_fwrite, ptr, size, nitems, stream)
+Z_SYSCALL_HANDLER(zephyr_fwrite, ptr, size, nitems, stream)
 {
 
 	Z_OOPS(Z_SYSCALL_MEMORY_ARRAY_READ(ptr, nitems, size));
-	return z_impl__zephyr_fwrite((const void *_MLIBC_RESTRICT)ptr, size,
+	return z_impl_zephyr_fwrite((const void *_MLIBC_RESTRICT)ptr, size,
 				    nitems, (FILE *_MLIBC_RESTRICT)stream);
 }
 #endif
@@ -94,7 +94,7 @@ Z_SYSCALL_HANDLER(z_zephyr_fwrite, ptr, size, nitems, stream)
 size_t fwrite(const void *_MLIBC_RESTRICT ptr, size_t size, size_t nitems,
 			  FILE *_MLIBC_RESTRICT stream)
 {
-	return z_zephyr_fwrite(ptr, size, nitems, stream);
+	return zephyr_fwrite(ptr, size, nitems, stream);
 }
 
 

